relay-system/src/controller.rs

```diff
@@ -102,6 +108,22 @@ impl Controller {
         Controller::from_registry().do_send(Subscribe(addr.recipient()))
     }
 
+    /// Subscribes to the [`Shutdown`] message to handle graceful shutdown.
+    ///
+    /// Returns a receiver for the [`Shutdown`] message, to be used to gracefully
+    /// shutdown.  This sends a message to the [`Controller`] actor so will
+    /// block until this actor is running.
+    ///
+    /// TODO(tobias): The receiver of this message can not yet signal they have completed
+    /// shutdown.
+    pub async fn subscribe_v2() -> watch::Receiver<Option<Shutdown>> {
```

Comment:
I feel like this is a weird custom wrapper around sending a message.  Let's leave this for this PR but maybe follow up with a small refactor that simply removes this wrapper and lets other actors directly send the message.
/cc @jan-auer

Reply:
It is weird, though the SubscribeV2 message will soon go away as we will also make the Controller a non-Actor. You can then simply call a method to get the receiver. 

In an offline conversation with @tobias-wilfert, we discussed two potential options:
 - Evolve this into a non-async method that returns the receiver
 - Make it an async function that resolves exactly once when the signal is sent. The `watch` then becomes a full implementation detail and all other services simply have to await this method. 

I'm not entirely sure if the second version would work. We can follow up with that in a separate PR.